mcfly/js/stores/CartStore.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var McFly = require('../dispatcher/McFly');
+var ProductStore = require('./ProductStore');
+var WebAPIUtils = require('../utils/WebAPIUtils');
+var assign = require('object-assign');
+
+var _products = {};
+
+function _addToCart (product) {
+    var id = product.id;
+    product.quantity = id in _products ? _products[id].quantity + 1 : 1;
+    _products[id] = assign({}, product[id], product);
+}
+
+var CartStore = McFly.createStore({
+    getAddedProducts: function () {
+        return Object.keys(_products).map(function (id) {
+            return _products[id];
+        });
+    },
+
+    getTotal: function () {
+        var total = 0;
+        for (var id in _products) {
+            var product = _products[id];
+            total += product.price * product.quantity;
+        }
+        return total.toFixed(2);
+    },
+}, function(payload) {
+
+    switch (payload.actionType) {
+        case 'ADD_TO_CART':
+            McFly.dispatcher.waitFor([ProductStore.dispatcherID]);
+            _addToCart(payload.product);
+            CartStore.emitChange();
+            break;
+        case 'CART_CHECKOUT':
+            WebAPIUtils.checkoutProducts(payload.products);
```

Comment:
@kenwheeler I prefer to not call APIs within stores.. is it possible to call `WebAPIUtils` in related action somewhere - https://github.com/voronianski/flux-comparison/pull/7/files#diff-0ae9121052ff6bc4e77cab865e4976feR18